include/boost/gil/channel_algorithm.hpp

```diff
@@ -333,14 +333,14 @@ template <> struct channel_convert_to_unsigned<bits8s> {
     typedef bits8s argument_type;
     typedef bits8 result_type;
     typedef bits8 type;
-    type operator()(bits8s  val) const { return val+128; }
+    type operator()(bits8s  val) const { return static_cast<bits8>(val+128); }
```

Comment:
Precisely, do you mean `static_cast<bits8>(val)+128` ?

AFAIU, since `val` is of type smaller than `int`, it is first promoted to `int`, regardless of signedness.
The `128` is `int`.
The two `int` are added.
Finally, result is truncated to fit into/stored in `unsigned char` type.
There should be no unexpected results in this particular case, so the cast just makes the result conversion explicit.

Please, correct me if I'm wrong.

Reply:
> `INT8_MAX+1`

Better, `std::numeric_limits<std::int8_t>::max` or similar, but no macros :)

Reply:
+1 on no macros. But also: let's not over-engineer this. Code readability is important, too !
Sometimes a little inline comment goes a long ways to make the intent explicit (if in doubt).

Reply:
I'm just raising this because we fixed a bug in gil via:
`type operator()(bits32s x) const { return static_cast<bits32>(x)+(static_cast<bits32>(1)<<31); }`

Reply:
This feels related.

Reply:
Yes, correct. But here we have 32-bit integers plus potential for arithmetic overflow, so that is subtly different situation.

Perhaps, to make it clear for the code reader that all the subtleties have been considered, we could fix it this way:

```
type operator()(bits8s  val) const { return static_cast<bits8>(static_cast<bits32>(val)+128u); }
```

What you think?

Reply:
That works for me.

Reply:
Good. Refined this PR in #76